avocado/core/test.py

```diff
@@ -606,6 +611,14 @@ def _check_reference_stdout(self):
                 os.path.isfile(self._expected_stdout_file)):
             expected = genio.read_file(self._expected_stdout_file)
             actual = genio.read_file(self._stdout_file)
+
+            diff = unified_diff(expected.splitlines(), actual.splitlines(),
+                                fromfile=self._expected_stdout_file,
+                                tofile=self._stdout_file)
+            with open(self._stdout_diff_file, 'w') as file_obj:
+                for line in diff:
+                    file_obj.write(line.strip()+'\n')
```

Comment:
how about `file_obj.writelines()`? 😉

Reply:
Did you test it? ;)

Reply:
Not here, but in interpret it worked well (using difflib)

Reply:
Nope, it do not.

Reply:
OK, because of the missing `\n`s, right? Well then either what you did or `file_obj.writelines(_ + '\n' for _ in diff)` 😄